Source/Urho3D/Resource/JSONValue.cpp

```diff
@@ -497,12 +498,32 @@ void JSONValue::SetVariantValue(const Variant& variant, Context* context)
         }
         return;
 
+    case VAR_CUSTOM_STACK:
+        {
+            SetType(JSON_NULL);
+            if (const auto* ptr = variant.GetCustomPtr<SharedPtr<Serializable>>())
+            {
+                if (ptr->NotNull())
+                {
+                    Serializable* object = ptr->Get();
+                    SetType(JSON_OBJECT);
+                    (*this)["type"] = object->GetTypeName();
+                    JSONValue& value = (*this)["value"];
```

Comment:
You can create and de-serialize JSONValue on stack (unlike XML) so u don't need these back-and-forth switches.
Just de-serialize everything _and then_ setup `this` to final value.